### PR TITLE
ISPN-10174 CLI Batch mode

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-10714

### CODE REVIEW

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            }
+         }
+
+         if (commandName == null)
+            commandName = "batch";
+         StringBuilder sb = new StringBuilder(commandName);
+         if (args.length > 0) {
+            sb.append(" ");
+            if (args.length == 1) {
+               sb.append(args[0]);
```
Comment (pruivo) [inline cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java]: should `args[0]` be quoted?

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+               registryBuilder.command(command);
+               registryBuilder.commands(subCommands);
+               runtime = AeshCommandRuntimeBuilder.builder().commandRegistry(registryBuilder.create()).build();
+            }
+         }
+
+         if (commandName == null)
+            commandName = "batch";
+         StringBuilder sb = new StringBuilder(commandName);
+         if (args.length > 0) {
```
Comment (pruivo) [inline cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java]: wouldn't be simpler to do:
```
         for (String arg : args) {
            sb.append(" \"").append(arg).append('"');
         }
```

File: cli/cli-client/src/main/java/org/infinispan/cli/CLI.java [hunk 3]
```diff
@@ -270,8 +282,6 @@ private void help(PrintStream out) {
       out.printf("Usage: cli [OPTION]...\n");
       out.printf("  -c, --connect=URL         %s\n", MSG.cliHelpConnect(Version.getBrandName()));
       out.printf("                            %s\n", MSG.cliHelpConnectHTTP());
-      out.printf("                            %s\n", MSG.cliHelpConnectJMXRMI());
-      out.printf("                            %s\n", MSG.cliHelpConnectJMXRemoting());
       out.printf("  -f, --file=FILE           %s\n", MSG.cliHelpFile());
```
Comment (pruivo) [inline cli/cli-client/src/main/java/org/infinispan/cli/CLI.java]: "If FILE is '-', then commands will be read from stdin."
I was unable to make it work. is it supported?

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import org.aesh.command.CommandNotFoundException;
+import org.aesh.command.CommandRuntime;
+import org.aesh.command.impl.registry.AeshCommandRegistryBuilder;
+import org.aesh.command.invocation.CommandInvocation;
+import org.aesh.command.parser.CommandLineParserException;
+import org.aesh.command.registry.CommandRegistryException;
+import org.aesh.command.validator.CommandValidatorException;
+import org.aesh.command.validator.OptionValidatorException;
+
+public class CliRuntimeRunner {
```
Comment (pruivo) [inline cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java]: is this class be used for something more than batching? 
It seems to have more stuff in it then it needs.

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java (hunk 1 above)
Comment (tristantarrant) [reply cli/cli-client/src/main/java/org/infinispan/cli/impl/CliRuntimeRunner.java]: No, commands are always single words

**Pull request merged**